tasks/run_puppet_device.rb

```diff
@@ -69,9 +69,7 @@ def read_device_certificate_fingerprints(cert_name)
 # Run 'puppet device' for each device, or just the target device.
 
 def run_puppet_device(devices, noop, timeout, apply)
-  os = Facter.value(:os) || {}
-  osfamily = os['family']
-  if osfamily == 'windows'
+  if Gem.win_platform?
```

Comment:
I'm surprised that this change is necessary (I developed in PE) ... but maybe I forget or am wrong.
In any case, did you consider using one of Puppet methods: 

- Puppet.features.microsoft_windows?
- Puppet::Util::Platform.windows?

https://tickets.puppetlabs.com/browse/PUP-8488